storage: bump Pebble and update interval collector implementation

RaduBerinde opened the pull request:
This change updates the interval collector implementation to match the
simplified interface in Pebble. We now have a stateless object that
just maps keys and ranges to timestamp intervals.

Pebble change:

 * [`654324f9`](https://github.com/cockroachdb/pebble/commit/654324f9) sstable: improve the property collector interface

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127519)
<!-- Reviewable:end -->

jbowens reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127519#-:-O2BPX5c2pPXcEJWEoU9:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
storage: bump Pebble and update interval collector implementation

This change updates the interval collector implementation to match the
simplified interface in Pebble. We now have a stateless object that
just maps keys and ranges to timestamp intervals.

Pebble change:

 * [`654324f9`](https://github.com/cockroachdb/pebble/commit/654324f9) sstable: improve the property collector interface

Epic: none
Release note: None

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127519#-:-O2BdFtkBWQXUha9PBIh:b-814g8k)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
